samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -153,7 +152,8 @@ public ClusterBasedJobCoordinator(Config coordinatorSystemConfig) {
     config = jobModelManager.jobModel().getConfig();
     hasDurableStores = new StorageConfig(config).hasDurableStores();
     state = new SamzaApplicationState(jobModelManager);
-    partitionMonitor = getPartitionCountMonitor(config);
+    systemAdmins = new SystemAdmins(config);
```

Comment:
Let's add the same comment as above: Should be started/stopped before ...

Reply:
Sure. I added comment `The systemAdmins should be started before partitionMonitor can be used. And it should be stopped when this coordinator is stopped.`